src/python/m5/util/convert.py

```diff
@@ -118,6 +133,32 @@ def _split_suffix(value, suffixes):
     return (value[: -len(matches[0])], matches[0]) if matches else (value, "")
 
 
+def _split_suffix_by_prefix(value):
```

Comment:
I think there should be a way of just having one `_split_suffix` function but i'm getting very confused looking at this files and how it splits these string so am fine with having two functions for now.

Reply:
In fact, doesn't `_split_suffix(value, suffixes=binary_prefixes)` do this for you?

Reply:
_split_suffix only matches the ending, so there would need to be two additional lists for the B and the B/s endings, e.g. [kB, MB, ...] and [kB/s, MB/s,...]. This seemed worse to me than adding a helper function, but is it more preferable?

Reply:
It occurred to me that I could put values through _split_suffix twice instead of using _split_suffix_by_prefix. I think I am going to refactor the code so it does this.

Reply:
It occurred to me that I could put values through _split_suffix twice instead of using _split_suffix_by_prefix. I think I am going to refactor the code so it does this.